libs/thread_data/src/execution_agent.cpp

```diff
@@ -128,7 +125,9 @@ namespace hpx { namespace threads {
         thread_id_type id = self_.get_thread_id();
 
         // handle interruption, if needed
-        threads::interruption_point(id);
+        thread_data* thrd_data = get_thread_id_data(id);
```

Comment:
Should we remove the include `thread_data.hpp` with the use of `get_thread_id_data` ?

Reply:
Most likely not. Good catch!